extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -277,24 +279,24 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
       this.element,
       'amp-story-page'
     );
-
     this.tagData_ =
+      shoppingData[pageElement.id] &&
       shoppingData[pageElement.id][
         this.element.getAttribute('data-product-id')
       ];
     if (this.hasAppendedInnerShoppingTagEl_ || !this.tagData_) {
       return;
     }
-
-    this.onRtlStateUpdate_(this.storeService_.get(StateProperty.RTL_STATE));
     this.shoppingTagEl_ = this.renderShoppingTagTemplate_();
-
     createShadowRootWithStyle(
       this.element,
       this.shoppingTagEl_,
       shoppingSharedCSS + shoppingTagCSS
     );
     this.hasAppendedInnerShoppingTagEl_ = true;
     this.styleTagText_();
+    this.onRtlStateUpdate_(this.storeService_.get(StateProperty.RTL_STATE));
+    this.flipTagIfOffscreen_(this.storeService_.get(StateProperty.PAGE_SIZE));
+    this.toggleShoppingTagActive_(pageElement.id);
```

Comment:
We could instantiate the tag specific listeners here instead of calling these methods and doing null checks within the methods. 
Check out the example in the Conversation tab.

Reply:
yes, sounds good, I have removed the null checks and only instantiate when nesc.